### PR TITLE
feat: remove thumbnails across mobile app

### PR DESCRIPTION
**Pull request by hlkavya0213**
Screenshot 2023-10-09 at 1 16 26 PM

Screenshot 2023-10-09 at 1 10 25 PM

image

### <samp>🤖 Generated by Copilot at 3eedd76</samp>

This pull request refactors and updates the code related to the receipt thumbnail functionality in the mobile app. It replaces the deprecated async function with waitForAsync in the spec files, removes unused code and properties from the file service and the file object model, and changes the receipt thumbnail element from a div to an image in the expense card components. It also improves the code quality and consistency by fixing linting errors and removing trailing commas.

### <samp>🤖 Generated by Copilot at 3eedd76</samp>

> _No more purpose, no more thumbnail_
> _We purge the code of obsolete trash_
> _We refactor, we simplify, we waitForAsync_
> _We are the masters of the file service class_

### <samp>🤖 Generated by Copilot at 3eedd76</samp>

*  Remove the purpose property from the file object model and the mock data, as it is no longer used in the file service or the components ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-b5604afda033e8ee07244ac5848af54d72cfd93a0fefa253923914e0a68346c8L84), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-25d328b14c02ccc965f1765483c032efb15db0f256b7c69479ac10b5c517621eL24-L42), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L8-R11), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-37b3100e61775b77a89941c8d30f7ff7c524a097b2a69c7ee082875f4246f540L2-L3), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL23-R23), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL29-R32))
*  Remove the downloadThumbnailUrl and getFilesWithThumbnail methods from the file service, as they are no longer used and rely on the purpose property ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-25d328b14c02ccc965f1765483c032efb15db0f256b7c69479ac10b5c517621eL24-L42))
*  Remove the test cases for the downloadThumbnailUrl and getFilesWithThumbnail methods from the file service spec, as they are no longer relevant ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-28b7792e6a56395a08128507b5f44cf54ad6bf9464d99e02865a88f129d9d017L45-L77))
*  Replace the receiptThumbnail property with the isReceiptPresent property in the expense card lite and expenses card components, as the receiptThumbnail property is no longer used and the isReceiptPresent property is a boolean flag that indicates whether the expense has a receipt or not ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-253227a5caba9ad01963ab55d217b355b5f906f7d66a9e0a2ec79f5f72621312L3-R3), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-37b3100e61775b77a89941c8d30f7ff7c524a097b2a69c7ee082875f4246f540L15-R13), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-e10e3621d78ca0dffb9376eb85cd98d0545ea527d78e492ac3b2d34fc901be52L29-R29), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL104-R104))
*  Simplify the getReceipt method in the expense card lite and expenses card components, as it only needs to call the findByTransactionId method of the file service and set the isReceiptPresent property based on the length of the files array returned by the method ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-37b3100e61775b77a89941c8d30f7ff7c524a097b2a69c7ee082875f4246f540L24-R24), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL170-R170))
*  Remove the test cases for the getReceipt method from the expenses card component spec, as they are no longer relevant ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL227-L289))
*  Replace the receipt thumbnail container div with an image element in the expense card lite and expenses card component templates, as the receipt thumbnail container div used a background image style that is no longer available and the image element uses a static placeholder image for the receipt ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-253227a5caba9ad01963ab55d217b355b5f906f7d66a9e0a2ec79f5f72621312L12-R14), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-e10e3621d78ca0dffb9376eb85cd98d0545ea527d78e492ac3b2d34fc901be52L52-R58))
*  Replace the inlineReceiptDataUrl and the thumbnail property of the tx_dataUrls array with a static placeholder image in the expenses card component template, as they are no longer available and the placeholder image is used for the receipt when the attachment upload is in progress or the expense is in the outbox ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-e10e3621d78ca0dffb9376eb85cd98d0545ea527d78e492ac3b2d34fc901be52L65-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-e10e3621d78ca0dffb9376eb85cd98d0545ea527d78e492ac3b2d34fc901be52L80-R83), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-e10e3621d78ca0dffb9376eb85cd98d0545ea527d78e492ac3b2d34fc901be52L94-R97))
*  Rename the receipt thumbnail container class to receipt image container class and add a receipt image class in the expense card lite and expenses card component scss files, as they reflect the change in the template and the use of an image element instead of a div ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d824aed5886c5eac27f33cb58094437fc1a4a33f2eac2a1f828506e49dfa27aeL31-R31), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d824aed5886c5eac27f33cb58094437fc1a4a33f2eac2a1f828506e49dfa27aeR41-R45), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-38009d2e072cea236cbced7bdb3d435ca6c18e445ecac6b68e88700414be5ae9L112-R112), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-38009d2e072cea236cbced7bdb3d435ca6c18e445ecac6b68e88700414be5ae9R122-R126))
*  Remove the setThumbnail method from the expenses card component, as it is no longer used and relies on the file service methods that are no longer used ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL456-L465))
*  Remove the test cases for the setThumbnail method from the expenses card component spec, as they are no longer relevant ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL899-L924))
*  Replace the setThumbnail method call with the isReceiptPresent property assignment in the addAttachments method in the expenses card component, as the setThumbnail method is no longer used and the isReceiptPresent property is a boolean flag that indicates whether the expense has a receipt or not ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL485-R449))
*  Remove the unused imports, trailing commas, and redundant assignments from various files, as they are not needed and may cause linting errors or reduce readability ([link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-28b7792e6a56395a08128507b5f44cf54ad6bf9464d99e02865a88f129d9d017L3-R3), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-25d328b14c02ccc965f1765483c032efb15db0f256b7c69479ac10b5c517621eL15-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-25d328b14c02ccc965f1765483c032efb15db0f256b7c69479ac10b5c517621eL54-R32), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-25d328b14c02ccc965f1765483c032efb15db0f256b7c69479ac10b5c517621eL62-R40), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-253227a5caba9ad01963ab55d217b355b5f906f7d66a9e0a2ec79f5f72621312L32-R28), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L27-R18), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L45-L47), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L54-R68), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-c510fadba8e4be58bb1b5e4db3b8ed6ad9f2bb5a0510742d369992da89751a35L96-R75), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL46-L53), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL82-L83), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL92-L93), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL177), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL745), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-a64806c6ea4e8f985ab1986cae8e8579932a34382d4dfc6b6e97bccac9d4c85eL754), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL138-R138), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL284-R255), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL304-R275), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL310-R281), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL318-R289), [link](https://github.com/fylein/fyle-mobile-app/pull/2499/files?diff=unified&w=0#diff-d44431e317039b0c96826f548a68ff36ebdd3c0c7f105ccb1ebf4b0b5546d18cL495-R456))

## Clickup
Please add link herehttps://app.clickup.com/t/85zu2xh64

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/file.service.spec.ts [hunk 1]
```diff
@@ -42,39 +36,6 @@ describe('FileService', () => {
     expect(fileService).toBeTruthy();
   });
 
-  it('downloadThumbnailUrl(): should return the file obj with thumbnail url', (done) => {
```
Comment (hlkavya0213) [inline src/app/core/services/file.service.spec.ts:45]: removed these methods

File: src/app/shared/components/expenses-card/expenses-card.component.spec.ts [hunk 2]
```diff
@@ -224,69 +205,6 @@ describe('ExpensesCardComponent', () => {
   });
 
   describe('getReceipt', () => {
-    it('should get the receipts when the file ids are present and the length of the thumbnail files array is greater that 0', fakeAsync(() => {
```
Comment (hlkavya0213) [inline src/app/shared/components/expenses-card/expenses-card.component.spec.ts:227]: these are not needed

File: src/app/shared/components/expenses-card/expenses-card.component.spec.ts [hunk 3]
```diff
@@ -896,32 +812,6 @@ describe('ExpensesCardComponent', () => {
     }));
   });
 
-  describe('setThumbnail():', () => {
```
Comment (hlkavya0213) [inline src/app/shared/components/expenses-card/expenses-card.component.spec.ts:899]: not needed

File: src/app/core/mock-data/file-object.data.ts [hunk 4]
```diff
@@ -81,7 +81,6 @@ export const fileObjectData5: FileObject = {
 export const thumbnailUrlMockData: FileObject[] = [
```
Comment (abhishek1234321) [inline src/app/core/mock-data/file-object.data.ts:81]: why do we still need this thumbnailUrlMockData?

**Review by abhishek1234321 — APPROVED** [review]

File: src/app/core/mock-data/file-object.data.ts (hunk 4 above)
Comment (hlkavya0213) [reply src/app/core/mock-data/file-object.data.ts:81]: forgot to rename, it's a file mock response

File: src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts [hunk 5]
```diff
@@ -42,58 +31,48 @@ describe('ExpenseCardLiteComponent', () => {
...
+    it('should set isReceiptPresent to false when no files are present', () => {
+      initialSetup([]);
+      expect(fileService.findByTransactionId).toHaveBeenCalledWith('txn1234');
+      expect(expenseCardLiteComponent.isReceiptPresent).toBeFalsy();
     });
   });
 
-  it('should display the thumbnail when available', () => {
-    fixture.detectChanges();
+  it('should display the recipt when available', () => {
```
Comment (OmkarJ13) [inline src/app/shared/components/expense-card-lite/expense-card-lite.component.spec.ts:60]: recipt -> receipt typo

File: src/assets/images/pdf_receipt_placeholder.png
Comment (OmkarJ13) [inline src/assets/images/pdf_receipt_placeholder.png:1]: I guess we are following the `-` way for assets in mobile app, let's rename this to `pdf-receipt-placeholder.png` to follow the same?

**Review by OmkarJ13 — APPROVED** [review]
2 comments

**Pull request merged**